Xamarin.Forms.Core/MessagingCenter.cs

```diff
@@ -91,7 +91,7 @@ static void InnerSend(string message, Type senderType, Type argType, object send
 			List<Tuple<WeakReference, Action<object, object>>> actionsCopy = actions.ToList();
 			foreach (Tuple<WeakReference, Action<object, object>> action in actionsCopy)
 			{
-				if (action.Item1.IsAlive && actions.Contains(action))
+				if (action.Item1.Target != null && actions.Contains(action))
 					action.Item2(sender, args);
```

Comment:
Btw, it appears that the tuple decouples the subscriber from the action; however, we check to see if the subscriber is alive. I'm not sure if we want to or expect to see the subscriber stick around until the action completes. So an adjustment to the loop would look like this:

```
object subscriber = action.Item1.Target;

if(subscriber == null || !actions.Contains(action))
   continue;

action.Item2(sender, args);
GC.KeepAlive(subscriber);
```

Just a thought. Obviously, if the action had a strong reference to the subscriber, the GC would keep it alive, but this is just enforcing pseudo parent-child relationship.